src/client/fontengine.cpp

```diff
@@ -439,59 +324,35 @@ void FontEngine::initSimpleFont(unsigned int basesize, FontMode mode)
 			m_settings->getFloat("gui_scaling") *
 			basesize);
 
-	irr::gui::IGUIFont* font = NULL;
-
-	for(unsigned int offset = 0; offset < MAX_FONT_SIZE_OFFSET; offset++) {
-
-		// try opening positive offset
-		std::stringstream fontsize_plus_png;
-		fontsize_plus_png << basename << "_" << (size + offset) << ".png";
-
-		if (fs::PathExists(fontsize_plus_png.str())) {
-			font = m_env->getFont(fontsize_plus_png.str().c_str());
-
-			if (font) {
-				verbosestream << "FontEngine: found font: " << fontsize_plus_png.str() << std::endl;
-				break;
-			}
-		}
+	irr::gui::IGUIFont *font = nullptr;
+	std::string font_extensions[] = { ".png", ".xml" };
 
-		std::stringstream fontsize_plus_xml;
-		fontsize_plus_xml << basename << "_" << (size + offset) << ".xml";
+	// Find nearest matching font scale
+	// Does a "zig-zag motion" (positibe/negative), from 0 to MAX_FONT_SIZE_OFFSET
```

Comment:
Also same functionality here. Positive offset it checked first, then goes from 0 to -MAX_FONT_SIZE_OFFSET and +MAX_FONT_SIZE_OFFSET